src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -245,6 +245,11 @@ private void AddOriginToResult(string origin, CorsPolicy policy, CorsResult resu
             else if (policy.Origins.Contains(origin))
             {
                 result.AllowedOrigin = origin;
+
+                if(policy.Origins.Count > 1)
```

Comment:
Interestingly, the [MDN doc](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS) has been modified to say that 

> If the server specifies an origin host rather than "*", **then it could also include Origin in the Vary response header** to indicate to clients that server responses will differ based on the value of the Origin request header.

So it isn't required.